portal/src/graphql/portal/BotProtectionConfigurationScreen.tsx

```diff
@@ -0,0 +1,449 @@
+import { Context, FormattedMessage } from "@oursky/react-messageformat";
+import React, { useCallback, useContext, useState } from "react";
+import cn from "classnames";
+import ScreenContent from "../../ScreenContent";
+import ScreenTitle from "../../ScreenTitle";
+import styles from "./BotProtectionConfigurationScreen.module.css";
+import {
+  BotProtectionProviderType,
+  PortalAPIAppConfig,
+  PortalAPISecretConfig,
+  PortalAPISecretConfigUpdateInstruction,
+} from "../../types";
+import {
+  AppSecretConfigFormModel,
+  useAppSecretConfigForm,
+} from "../../hook/useAppSecretConfigForm";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import { useAppSecretVisitToken } from "./mutations/generateAppSecretVisitTokenMutation";
+import { AppSecretKey } from "./globalTypes.generated";
+import ShowError from "../../ShowError";
+import ShowLoading from "../../ShowLoading";
+import { useLocationEffect } from "../../hook/useLocationEffect";
+import { produce } from "immer";
+import { clearEmptyObject } from "../../util/misc";
+import FormContainer from "../../FormContainer";
+import ScreenDescription from "../../ScreenDescription";
+import Toggle from "../../Toggle";
+import WidgetTitle from "../../WidgetTitle";
+import { DefaultEffects, IButtonProps, Image, Label } from "@fluentui/react";
+import { useSystemConfig } from "../../context/SystemConfigContext";
+import recaptchaV2LogoURL from "../../images/recaptchav2_logo.svg";
+import cloudflareLogoURL from "../../images/cloudflare_logo.svg";
+import WidgetDescription from "../../WidgetDescription";
+import FormTextField from "../../FormTextField";
+import PrimaryButton from "../../PrimaryButton";
+import { startReauthentication } from "./Authenticated";
+
+const MASKED_SECRET = "***************";
+
+const SECRET_KEY_FORM_FIELD_ID = "secret-key-form-field";
+
+interface LocationState {
+  isOAuthRedirect: boolean;
+}
+function isLocationState(raw: unknown): raw is LocationState {
+  return (
+    raw != null &&
+    typeof raw === "object" &&
+    (raw as Partial<LocationState>).isOAuthRedirect != null
+  );
+}
+
+interface FormState {
+  enabled: boolean;
+  providerType: BotProtectionProviderType;
+  siteKey: string;
+  secretKey: string | null;
+}
```

Comment:
As mentioned in yesterday's product meeting, please separate the states of different providers.

```typescript
interface  CloudflareConfigs {
  siteKey: string;
  secretKey: string | null;
}

interface Recaptchav2Configs {
  siteKey: string;
  secretKey: string | null;
}

interface FormState {
    enabled: boolean;
    providerType: BotProtectionProviderType;
    providerConfigs: Record<BotProtectionProviderType, CloudflareConfigs | Recaptchav2Configs>
}
```